include/spin1_api.h

```diff
@@ -385,20 +405,64 @@ __forceinline uint spin1_int_disable(void)
 * SOURCE
 */
 #ifdef THUMB
+// Implemented in sark_alib.s; needs to be in ARM mode, not THUMB
 extern void spin1_mode_restore(uint cpsr);
-#elif defined(__GNUC__)
-__inline __attribute__((always_inline)) void spin1_mode_restore(uint cpsr)
+#else
+SPIN1_INLINE void spin1_mode_restore(uint cpsr)
 {
+    spin1_store_barrier();
+
+#ifdef __GNUC__
     asm volatile (
     "msr	cpsr_c, %[cpsr]"
     :
     : [cpsr] "r" (cpsr)
     :);
+#else
+    __asm { msr cpsr_c, sr }
```

Comment:
```
armcc -c --c99 --cpu=5te --apcs interwork --min_array_alignment=4 -I /c/Users/zzalsar4/git/spinnaker_tools/include --split_sections -g -Ospace -o ../build/arm/spin1_isr.o spin1_isr.c
"c:/Users/zzalsar4/git/spinnaker_tools/include/spin1_api.h", line 422: Error:  #20: identifier "sr" is undefined
      __asm { msr cpsr_c, sr }
                          ^
```

Reply:
Ah, should be cpsr I assume

Reply:
Version RVDS40 of armcc complains about the use of __cpsr__ in this instruction::

    "/spinnaker_tools/include/spin1_api.h", line 422: Error:  #1094: Expected a register expression
            __asm { msr cpsr_c, cpsr }
                                ^
    spin1_isr.c: 0 warnings, 1 error_

I suggest changing the parameter back to  __sr__  and updating the function prototype::

    SPIN1_INLINE void spin1_mode_restore(uint sr)

The GCC register map::

        : [cpsr] "r" (sr)

and the ARM instruction::

        __asm { msr cpsr_c, sr }

To get  rid of the compilation error.

Reply:
I think I have done this as requested:
https://github.com/SpiNNakerManchester/spinnaker_tools/pull/107/commits/6c6e6fdd6797d61d8e37de97154adc862a6997e7

Does this look correct?

Reply:
A merge with master changed the name to "value".  I am hoping this is now correct again

Reply:
The code no compiles correctly with ARM C/C++ Compiler, RVCT4.0 [Build 400].